### PR TITLE
Add Utf8String(string)

### PR DESCRIPTION
**Pull request by krwq**

### CODE REVIEW

File: src/System.Text.Utf8/src/System/Text/Utf16/Utf16LittleEndianEncoder.cs [hunk 1]
```diff
@@ -101,5 +101,28 @@ public static bool TryEncodeCodePoint(UnicodeCodePoint codePoint, ByteSpan buffe
             }
             return true;
         }
+
+        // TODO: Should we rewrite this to not use char.ConvertToUtf32 or is it fast enough?
+        public static bool TryDecodeCodePointFromString(string s, int index, out UnicodeCodePoint codePoint, out int encodedChars)
```
Comment (KrzysztofCwalina) [inline src/System.Text.Utf8/src/System/Text/Utf16/Utf16LittleEndianEncoder.cs]: This API should be on Utf8Encoder, not on Utf8String

Comment (krwq) [inline src/System.Text.Utf8/src/System/Text/Utf16/Utf16LittleEndianEncoder.cs]: I don't understand, It is on Utf16LE encoder, not on Utf8String or Utf8Encoder

Comment (KrzysztofCwalina) [inline src/System.Text.Utf8/src/System/Text/Utf16/Utf16LittleEndianEncoder.cs]: Ah, sorry. For some reason I thought it's on Utf8String.

**Comment by KrzysztofCwalina** [comment]
looks good!

File: src/System.Text.Utf8/src/System/Text/Utf16/Utf16LittleEndianEncoder.cs [hunk 2]
```diff
@@ -101,5 +101,28 @@ public static bool TryEncodeCodePoint(UnicodeCodePoint codePoint, ByteSpan buffe
...
+
+            if (index == s.Length - 1 && char.IsSurrogate(s[index]))
+            {
+                codePoint = default(UnicodeCodePoint);
+                encodedChars = 0;
+                return false;
+            }
+
+            encodedChars = char.IsHighSurrogate(s[index]) ? 2 : 1;
+            codePoint = (UnicodeCodePoint)(unchecked((uint)char.ConvertToUtf32(s, index)));
```
Comment (ellismg) [inline src/System.Text.Utf8/src/System/Text/Utf16/Utf16LittleEndianEncoder.cs]: This call can throw in the case where you have a high surrogate followed by another high surrogate, I think.

**Pull request merged**